graylog2-web-interface/src/hooks/useFavoriteItemMutation.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { useMutation } from '@tanstack/react-query';
+
+import fetch from 'logic/rest/FetchProvider';
+import { qualifyUrl } from 'util/URLUtils';
+import UserNotification from 'util/UserNotification';
+
+export const urlPrefix = '/startpage';
+
+const putFavoriteItem = async (id: string) => {
+  const url = `${urlPrefix}/addToFavorites/${id}`;
+
+  return fetch('PUT', qualifyUrl(url));
+};
+
+const deleteFavoriteItem = (id: string) => {
+  const url = `${urlPrefix}/removeFromFavorites/${id}`;
+
+  return fetch('DELETE', qualifyUrl(url));
+};
+
+const useFavoriteItemMutation = () => {
+  const putMutation = useMutation(putFavoriteItem, {
+    onSuccess: async () => {
+      UserNotification.success('Item added to favorite successfully.', 'Success');
```

Comment:
Let's use `favorites` instead of `favorite`.

Reply:
Maybe the color change of the fav icon is enough to communicate that the action has been successful and it is enough to display notification when the request failed?